packages/core-components/src/layout/DelegatedSignInPage/DelegatedSignInIdentity.ts

```
@@ -0,0 +1,199 @@
+          signalErrors = true;
+          pause = tokenToExpiryMillis(this.session?.backstageIdentity.token);
+          if (pause <= 0) {
+            this.session = undefined;
+          }
+        } catch {
+          signalErrors = false; // Only signal first failure in a row
+          pause = DEFAULTS.retryRefreshFrequencyMillis;
+        }
+      }
+    };
+
+    const refreshLoop = async () => {
+      while (!this.abortController.signal.aborted) {
+        await sleep(pause, this.abortController.signal);
+        await refreshOnce();
+      }
+    };
+
+    // Try first refresh and bubble up any errors to the caller, then get the
+    // loop going behind the scenes and return
+    await refreshOnce();
+    refreshLoop();
+  }
+
+  /** {@inheritdoc @backstage/core-plugin-api#IdentityApi.getUserId} */
+  getUserId(): string {
+    return this.getSession().backstageIdentity.id;
+  }
+
+  /** {@inheritdoc @backstage/core-plugin-api#IdentityApi.getIdToken} */
+  async getIdToken(): Promise<string | undefined> {
+    return this.getSession().backstageIdentity.token;
+  }
+
+  /** {@inheritdoc @backstage/core-plugin-api#IdentityApi.getProfile} */
+  getProfile(): ProfileInfo {
+    return this.getSession().profile;
+  }
+
+  /** {@inheritdoc @backstage/core-plugin-api#IdentityApi.getProfileInfo} */
+  async getProfileInfo(): Promise<ProfileInfo> {
+    return this.getSession().profile;
+  }
+
+  /** {@inheritdoc @backstage/core-plugin-api#IdentityApi.getBackstageIdentity} */
+  async getBackstageIdentity(): Promise<BackstageUserIdentity> {
+    return this.getSession().backstageIdentity.identity;
+  }
+
+  /** {@inheritdoc @backstage/core-plugin-api#IdentityApi.getCredentials} */
+  async getCredentials(): Promise<{ token?: string | undefined }> {
+    return { token: this.getSession().backstageIdentity.token };
+  }
+
+  /** {@inheritdoc @backstage/core-plugin-api#IdentityApi.signOut} */
+  async signOut(): Promise<void> {
+    this.abortController.abort();
```

I wasn't sure what to actually do here. Should we just throw an exception? Should we refresh the page? Should we for real try to sign out of the session somehow?